### PR TITLE
Add required directories when using custom jails

### PR DESCRIPTION
**Pull request by danifr**
Fix for https://github.com/voxpupuli/puppet-fail2ban/issues/117

### CODE REVIEW

File: manifests/jail.pp [hunk 1]
```diff
@@ -29,6 +29,13 @@
 ) {
 
   # Jail filter creation
+  file { "$config_dir_filter_path":
+    ensure => directory,
+    owner  => $config_file_owner,
+    group  => $config_file_group,
+    before => File["custom_filter_${name}"]
```
Comment (traylenator) [inline manifests/jail.pp]: before and after for files and directories are not needed, it's automatic.

File: manifests/jail.pp [hunk 2]
```diff
@@ -52,6 +59,13 @@
   }
 
   # Jail creation
+  file { "${fail2ban::params::config_dir_path}/jail.d":
+    ensure => directory,
+    owner  => $config_file_owner,
+    group  => $config_file_group,
+    before => File["custom_jail_${name}"]
```
Comment (traylenator) [inline manifests/jail.pp]: before and after for files and directories are not needed, it's automatic.

File: manifests/jail.pp [hunk 3]
```diff
@@ -52,6 +59,13 @@
...
+  }
+
   file { "custom_jail_${name}":
     ensure  => file,
     path    => "${fail2ban::params::config_dir_path}/jail.d/${name}.conf",
```
Comment (traylenator) [inline manifests/jail.pp]: That reference to params is not right, pass it in from init like for instance `$config_file_owner`

Comment (traylenator) [reply manifests/jail.pp]: Sorry - that's not your code, the module is already wrong here :-)

**Comment by dhoppe** [comment]
I think this pull request can be closed, since all supported operating systems will provide these directories by default.

**Pull request closed without merging**